pkg/handlers/generic/lifecycle/clusterautoscaler/strategy_crs.go

```diff
@@ -119,3 +130,37 @@ func (s crsStrategy) apply(
 
 	return nil
 }
+
+func (s crsStrategy) delete(
+	ctx context.Context,
+	cluster *clusterv1.Cluster,
+	log logr.Logger,
+) error {
+	// The cluster-autoscaler is different from other addons.
+	// It requires all resources to be created in the management cluster,
+	// which means creating the ClusterResourceSet always targeting the management cluster.
+	targetCluster, err := findTargetCluster(ctx, s.client, cluster)
+	if err != nil {
+		return err
+	}
+
+	crs := &crsv1.ClusterResourceSet{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: crsv1.GroupVersion.String(),
+			Kind:       "ClusterResourceSet",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: targetCluster.Namespace,
+			Name:      s.config.defaultClusterAutoscalerConfigMap + "-" + cluster.Name,
```

Comment:
nit: it was kind of difficult to track down why the name is based on `defaultClusterAutoscalerConfigMap`, maybe some helper function to get the name would be useful?

https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/blob/eabc6896e1126086474a11f3153cb682cf427eb4/pkg/handlers/generic/lifecycle/clusterautoscaler/strategy_crs.go#L55